### PR TITLE
Enable Disable Story

### PR DESCRIPTION
**Pull request by mpeyfuss**
Added the ability to enable/disable story additions after the contract is created.

### CODE REVIEW

File: src/StoryContract.sol [hunk 1]
```diff
@@ -19,29 +19,67 @@ pragma solidity 0.8.17;
...
+error TokenDoesNotExist();
+
+/// @dev caller is not the token owner
+error NotTokenOwner();
+
+/// @dev caller is not the token creator
+error NotTokenCreator();
+
+/// @dev caller is not a story admin
+error NotStoryAdmin();
```
Comment (koloz193) [inline src/StoryContract.sol:41]: any reason to make these not part of the contract/interface like the events?

File: src/StoryContract.sol [hunk 2]
```diff
@@ -19,29 +19,67 @@ pragma solidity 0.8.17;
...
 
-    ///////////////////// STORY FUNCTIONS /////////////////////
+    /*//////////////////////////////////////////////////////////////////////////
+                                Story Functions
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @dev function to set story enabled/disabled
+    /// @dev requires story admin
+    function setStoryEnabled(bool enabled) external {
+        if (!_isStoryAdmin(msg.sender)) { revert NotStoryAdmin(); }
```
Comment (koloz193) [inline src/StoryContract.sol:80]: not sure if it cuts down on bytecode size but can technically remove `{` and `}`

File: src/StoryContract.sol [hunk 3]
```diff
@@ -19,29 +19,67 @@ pragma solidity 0.8.17;
...
         if (!storyEnabled) { revert StoryNotEnabled(); }
         _;
     }
 
-    ///////////////////// CONSTRUCTOR /////////////////////
+    /*//////////////////////////////////////////////////////////////////////////
+                                Constructor
+    //////////////////////////////////////////////////////////////////////////*/
 
     /// @param enabled is a bool to enable or disable Story addition. This cannot be undone later.
```
Comment (koloz193) [inline src/StoryContract.sol:68]: comment update since its updateable

File: src/StoryContract.sol [hunk 4]
```diff
@@ -19,29 +19,67 @@ pragma solidity 0.8.17;
...
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @dev function to set story enabled/disabled
+    /// @dev requires story admin
+    function setStoryEnabled(bool enabled) external {
+        if (!_isStoryAdmin(msg.sender)) { revert NotStoryAdmin(); }
+        storyEnabled = enabled;
+    }
 
     /// @dev see { IStory.addCreatorStory }
```
Comment (koloz193) [inline src/StoryContract.sol:84]: can do `/// @inheritdoc IStory`

File: src/upgradeable/StoryContractUpgradeable.sol [hunk 5]
```diff
@@ -77,14 +119,18 @@ abstract contract StoryContractUpgradeable is Initializable, IStory, ERC165Upgra
     /// @dev function to check creatorship of a token
     function _isCreator(address potentialCreator, uint256 tokenId) internal view virtual returns (bool);
 
-    ///////////////////// ERC-165 OVERRIDE /////////////////////
+    /*//////////////////////////////////////////////////////////////////////////
+                                Overrides
+    //////////////////////////////////////////////////////////////////////////*/
 
     /// @dev see { ERC165.supportsInterface }
```
Comment (koloz193) [inline src/upgradeable/StoryContractUpgradeable.sol:126]: can do `/// @inheritdoc ERC165`

File: src/IStory.sol [hunk 6]
```diff
@@ -21,20 +21,6 @@
...
-/// @dev token does not exist
-error TokenDoesNotExist();
-
-/// @dev caller is not the token owner
-error NotTokenOwner();
-
-/// @dev caller is not the token creator
-error NotTokenCreator();
-
 interface IStory {
```
Comment (koloz193) [inline src/IStory.sol:24]: should we add the `@parm` natspec comments?

File: src/StoryContract.sol (hunk 2 above)
Comment (mpeyfuss) [reply src/StoryContract.sol:80]: that's true and looks cleaner

File: src/StoryContract.sol (hunk 3 above)
Comment (mpeyfuss) [reply src/StoryContract.sol:68]: oh yeah

File: src/StoryContract.sol (hunk 4 above)
Comment (mpeyfuss) [reply src/StoryContract.sol:84]: ohhh nice

File: src/StoryContract.sol (hunk 1 above)
Comment (mpeyfuss) [reply src/StoryContract.sol:41]: mainly was for pulling in during tests - much easier than having inside the contract

Comment (mpeyfuss) [reply src/StoryContract.sol:41]: had it in the interface file but still had to be outside the interface in order to import

**Review by koloz193 — APPROVED** [review]
ship it! 🚀

**Pull request merged**